src/System.Net.Sockets/tests/FunctionalTests/UnixDomainSocketTest.netcoreapp.cs

```diff
@@ -350,6 +351,63 @@ public void UnixDomainSocketEndPoint_InvalidPaths_Throws()
             Assert.Throws<ArgumentOutOfRangeException>(() => new UnixDomainSocketEndPoint(invalidLengthString));
         }
 
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public void UnixDomainSocketEndPoint_RemoteEndPointEqualsBindAddress(bool abstractAddress)
+        {
+            string serverAddress;
+            string clientAddress;
+            if (abstractAddress)
+            {
+                // abstract socket addresses are a Linux feature.
+                if (!RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+                {
+                    return;
+                }
+                // An abstract socket address starts with a zero byte.
+                serverAddress = '\0' + Guid.NewGuid().ToString();
+                clientAddress = '\0' + Guid.NewGuid().ToString();
+            }
+            else
+            {
+                serverAddress = GetRandomNonExistingFilePath();
+                clientAddress = GetRandomNonExistingFilePath();
+            }
+
+            try
+            {
+                using (Socket server = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified))
+                {
+                    server.Bind(new UnixDomainSocketEndPoint(serverAddress));
+                    server.Listen(1);
+
+                    using (Socket client = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified))
+                    {
+                        // Bind the client.
+                        client.Bind(new UnixDomainSocketEndPoint(clientAddress));
+                        client.Connect(new UnixDomainSocketEndPoint(serverAddress));
+                        using (Socket acceptedClient = server.Accept())
+                        {
+                            // Verify the client address on the server.
+                            EndPoint clientAddressOnServer = acceptedClient.RemoteEndPoint;
+                            Assert.True(string.CompareOrdinal(clientAddress, clientAddressOnServer.ToString()) == 0);
```

Comment:
Ah, we bind the client to validate that it matches what the server gets?

Reply:
Is this an argument for exposing a UnixDomainSocketEndPoint.Path property?  Ideally someone wouldn't rely on ToString for this information.  Are there cases where a server would want to see what address the client was bound to like this?

Reply:
> Ah, we bind the client to validate that it matches what the server gets?

Yes, this is a way of using the SocketAddres -> UnixDomainSocketEndPoint conversion. And checking we are using the proper address.

> Is this an argument for exposing a UnixDomainSocketEndPoint.Path property? Ideally someone wouldn't rely on ToString for this information. Are there cases where a server would want to see what address the client was bound to like this?

For this test, it is important to check the path. In real code, a client socket won't name itself. I think it's fine there is no Path property.

Somewhat related, I think it would be nice if ToString would prepend path with '@' if it is an abstract path. The 0 byte isn't printed so it easy to miss it. '@' is often used to denote abstract paths by Linux tools. wdyt?

Reply:
> I think it's fine there is no Path property.

Ok, thanks for the confirmation.

> I think it would be nice if ToString would prepend path with '@'

I'm fine with that if it's the defacto standard and @eerhardt's cool with it.

Reply:
That sounds fine with me, and may help someone in debugging at some point.